Tests/MuxUploadSDKTests/UploadManagerTests/UploadManagerTests.swift

```diff
@@ -0,0 +1,59 @@
+//
+//  UploadManagerTests.swift
+//
+
+import Foundation
+import XCTest
+
+@testable import MuxUploadSDK
+
+class UploadManagerTests: XCTestCase {
+
+    func testUploadManagerURLDeduplication() throws {
+
+        let uploadManager = UploadManager()
+
+        let chunkFileUploader = ChunkedFileUploader(
+            uploadInfo: UploadInfo(
+                id: UUID().uuidString,
+                uploadURL: URL(string: "https://www.example.com/upload")!,
+                videoFile: URL(string: "file://path/to/dummy/file/")!,
+                chunkSize: 8,
+                retriesPerChunk: 2,
+                optOutOfEventTracking: true
+            )
+        )
+
+        let upload = MuxUpload(
+            wrapping: chunkFileUploader,
+            uploadManager: uploadManager
+        )
+
+        let duplicateChunkFileUploader = ChunkedFileUploader(
+            uploadInfo: UploadInfo(
+                id: UUID().uuidString,
+                uploadURL: URL(string: "https://www.example.com/upload")!,
+                videoFile: URL(string: "file://path/to/dummy/file/")!,
+                chunkSize: 8,
+                retriesPerChunk: 2,
+                optOutOfEventTracking: true
+            )
+        )
+
+        let duplicateUpload = MuxUpload(
+            wrapping: duplicateChunkFileUploader,
+            uploadManager: uploadManager
+        )
+
+        upload.start(forceRestart: true)
+        duplicateUpload.start(forceRestart: true)
+
+        XCTAssertEqual(
```

Comment:
This test currently fails for me, @daytime-em any tips for where I'm going wrong?